src/cli/global/list.rs

```diff
@@ -1,9 +1,375 @@
+use crate::global::common::find_package_records;
+use crate::global::project::ParsedEnvironment;
+use crate::global::{EnvironmentName, ExposedName, Project};
 use clap::Parser;
+use fancy_display::FancyDisplay;
+use human_bytes::human_bytes;
+use indexmap::IndexMap;
+use itertools::Itertools;
+use miette::{miette, IntoDiagnostic};
+use pixi_config::{Config, ConfigCli};
+use pixi_consts::consts;
+use pixi_spec::PixiSpec;
+use rattler_conda_types::{PackageName, PackageRecord, PrefixRecord, Version};
+use serde::Serialize;
+use std::io::{stdout, Write};
+use std::str::FromStr;
+use thiserror::__private::AsDisplay;
 
 /// Lists all packages previously installed into a globally accessible location via `pixi global install`.
+///
+/// All environments:
+/// - Yellow: the binaries that are exposed.
+/// - Green: the packages that are explicit dependencies of the environment.
+/// - Blue: the version of the installed package.
+/// - Cyan: the name of the environment.
+///
+/// Per environment:
+/// - Green: packages are explicitly installed.
 #[derive(Parser, Debug)]
-pub struct Args {}
+pub struct Args {
+    /// Answer yes to all questions.
+    #[clap(short = 'y', long = "yes", long = "assume-yes")]
+    assume_yes: bool,
 
-pub async fn execute(_args: Args) -> miette::Result<()> {
-    todo!()
+    #[clap(flatten)]
+    config: ConfigCli,
+
+    /// The name of the environment to list.
+    #[clap(short, long)]
+    environment: Option<String>,
+
+    /// Sorting strategy for the package table
+    #[arg(long, default_value = "name", value_enum)]
+    pub sort_by: GlobalSortBy,
+}
+
+pub async fn execute(args: Args) -> miette::Result<()> {
+    let config = Config::with_cli_config(&args.config);
+    let project = Project::discover_or_create(args.assume_yes)
+        .await?
+        .with_cli_config(config.clone());
+
+    if let Some(environment) = args.environment {
+        let name = EnvironmentName::from_str(environment.as_str())?;
+        // Verify that the environment is in sync with the manifest and report to the user otherwise
+        if !project.environment_in_sync(&name).await? {
+            tracing::warn!("The environment '{}' is not in sync with the manifest, to sync run\n\tpixi global sync", name);
+        }
+        list_environment(project, &name, args.sort_by).await?;
+    } else {
+        // Verify that the environments are in sync with the manifest and report to the user otherwise
+        // TODO: Check sync logic
+        list_global_environments(project).await?;
+    }
+
+    Ok(())
+}
+
+/// Sorting strategy for the package table
+#[derive(clap::ValueEnum, Clone, Debug, Serialize)]
+pub enum GlobalSortBy {
+    Size,
```

Comment:
If I read the code correctly, this sorts by size of package.
My feeling is that size of the environment would be the better metric here.
Also, it would be good if the size of the environment would be reported by `pixi global list`, at least when ones sorts by size

Reply:
Ah the comment might be lacking, it means that the "environment specific package list gets sorted" like `pixi list` does. e.g. this only works in together with `--environment`.

Reply:
At the moment, this doesn't throw an error when used without specifying `--environment`

Reply:
Updated the command